bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/VariableDelayPoller.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.opengarage.internal;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Variable poller will schedule a task to run periodically, but will allow the
+ * schedule to be modified on-the-fly
+ *
+ * @author Tim Harper - Initial contribution
+ */
+public class VariableDelayPoller {
+    private final Logger logger = LoggerFactory.getLogger(VariableDelayPoller.class);
+    private ScheduledExecutorService scheduler;
+    private Supplier<Long> task;
+    private long defaultPollSeconds;
+    private AtomicLong lastThreadId = new AtomicLong(0);
+    private Future<?> currentFuture;
+
+    /**
+     * @param scheduler - Reference to the Java Scheduler instance
+     * @param task - task to invoke. Task should return a number of seconds indicating when the next task should be
+     *            called
+     * @param defaultPollSeconds - how long to set the next poll should task fail to return a value due to an uncaught
+     *            exception
+     */
+    public VariableDelayPoller(ScheduledExecutorService scheduler, Supplier<Long> task, long defaultPollSeconds) {
+        this.scheduler = scheduler;
+        this.defaultPollSeconds = defaultPollSeconds;
+        this.task = task;
+
+        startNextPoll(lastThreadId.get(), 1);
+    };
+
+    synchronized public void stop(boolean abortIfRunning) {
+        lastThreadId.getAndIncrement(); // prevent future tasks from being scheduled
+        this.currentFuture.cancel(abortIfRunning);
+    }
+
+    private void doPoll(long threadId) {
+        var nextPollDuration = this.defaultPollSeconds;
+        try {
+            nextPollDuration = task.get();
+        } catch (Exception e) {
+            logger.error("error occurred while invoking periodic task", e);
```

Comment:
Please do not use ERROR level, either DEBUG or WARN. Use WARN only if you are sure it could not flood the logs.